adding examples and making modifications (#1070)
MicrosoftDocs/iis-docs: iis/manage/configuring-security/using-dynamic-ip-restrictions.md

On my IIS level, I have below configuration:

From “Edit Feature Settings”:

![alt text](image.png)

![alt text](image-1.png)

From “Edit Dynamic Restriction Settings…”:

![alt text](image-2.png)

![alt text](image-3.png)

On my site level I have below configuration: 

![alt text](image-4.png)

![alt text](image-5.png)

This means that up to 2 simultaneous requests will be allowed to the site (due to number 2 set above), and a status code of 403 will be received for any other concurrent request (due to “Forbidden” action configured above) as can be seen in browser developer tools (I clicked “disable cache” to make it clear and avoid browser side caching):

![alt text](image-6.png)

Note: You might be wondering why do you see 3 allowed requests and not just two; that's because the first two requests were concurrently opened and once finished it happened that the third request reached the application, meaning the third request was the only one (hence only 1 concurrent requests) when it happened.
 
The “Allow” action configured above means that any IP address is okay to send requests and that the rules configured specify on which conditions these allowed IPs will be denied access.
 

> As you noticed, site level config of “Allow” overridden IIS’s level “Deny”, and site's level config of “Forbidden” overridden IIS's level config of “Not Found”, and hence 403 status code was sent instead of 404.

Reviewer: Curly quotes (“”) should be changed to straight quotes (") throughout.

 
If I however change site's level “Allow” to “Deny” then all IPs will be denied access regardless of whether they made 2 concurrent requests or otherwise:

![alt text](image-7.png)

### Example 4: Blocking of IP address based on number of concurrent requests & number of requests over time, using UI

Now let's change the configuration from Example 3 a bit on site level. IIS level is same as before in Example 3:

![alt text](image-8.png)

While site level configuration is now: 

![alt text](image-9.png)

> Now when making request from browser:

Reviewer: ```suggestion
Now when making a request from browser:
```

![alt text](image-10.png)

> Each colored line (example highlighted) represents the lifetime of one request. The rule specifies two conditions as you can see. This means that any condition that gets met first then the denial will be applied. The rule will allow only up to two requests within 200 msec, and any further requests within this timeframe will be denied. As can be seen, the third request was triggered almost 20 msec after the first one, meaning now we have more than two requests within the 200 msec frame and hence the third one will be denied, and so will be the 4th and 5th and 6th.  

Reviewer: ```suggestion
Each colored line (example highlighted) represents the lifetime of one request. The rule specifies two conditions as you can see. This means that any condition that gets met first then the denial will be applied. The rule will allow only up to two requests within 200 msec, and any further requests within this timeframe will be denied. As can be seen, the third request was triggered almost 20 msec after the first one, meaning now we have more than two requests within the 200 msec frame and hence the third one will be denied, and so will the 4th and 5th and 6th.  
```

 
2024-07-22 14:17:48 ::1 GET /bundles/jquery v=JVBM4Dk7eZ-fjWvmxvoCeVR5MAt_3YXn4K7MRdHsuR81 8090 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/126.0.0.0+Safari/537.36+Edg/126.0.0.0 http://localhost:8090/ 403 502 0 7
 
2024-07-22 14:17:48 ::1 GET /bundles/bootstrap v=X8gnNIYDSsAzCxLBuTaZy64JJqo9mzWM5GPsE4TJLNI1 8090 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/126.0.0.0+Safari/537.36+Edg/126.0.0.0 http://localhost:8090/ 403 502 0 7
 
2024-07-22 14:17:48 ::1 GET /favicon.ico - 8090 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/126.0.0.0+Safari/537.36+Edg/126.0.0.0 http://localhost:8090/ 403 502 0 0
 
2024-07-22 14:17:48 ::1 GET /Content/bootstrap.min.css.map - 8090 - ::1 Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/126.0.0.0+Safari/537.36+Edg/126.0.0.0 - 403 502 0 0
 

> Http status code for denied requests in this example will be 403 and sub status code will be 502. This way you can tell the difference between normal Forbidden status code set by the application code and between the Forbidden set by the IIS IP Restrictions feature.

Reviewer: ```suggestion
Http status code for denied requests in this example will be 403 and sub status code will be 502. This way you can tell the difference between normal Forbidden status code set by the application code and the Forbidden set by the IIS IP Restrictions feature.
```

List of (status codes) set by IP Restriction feature are discussed below.

### Deny Actions

The module can be configured to perform the following actions when denying requests for IP addresses:

•	Forbidden: 403
•	Unauthorized: 401
•	Not Found: 404
•	Abort: the TCP connection will simply be reset and http will not be established. This is also nice in case main page is denied and you don't want the end user to see the sub status code 502 (to not know this was because of IP Restriction feature). User will instead see the below: 
![alt text](image-11.png)

And here's the possible (substatus codes) by IP Restriction feature:
501, 502.
Substatus code will either be 501 or 502 depending on which one of the two conditions got met and hence client was denied access: 
501 -> Dynamic IP Restriction: too many concurrent requests were made from the same client IP.  
502 -> Dynamic IP Restriction: the maximum number of requests from the same client IP within a specified time limit was reached.
 

> Above codes are documented here: HTTP status code overview - Internet Information Services | Microsoft Learn

Reviewer: ```suggestion
The preceding codes are documented here: HTTP status code overview - Internet Information Services | Microsoft Learn
```
It looks like this should be a link.

## Enable Logging Only Mode: 

> Now in case of “Enable Logging Only Mode” there will be no clients denied by this feature, meaning status code will be the expected result (normally 200 but it also depends on your application) and clients will receive the expected normal response, however, substatus code indicating that one of the two conditions was met will be logged in IIS logs. Below is an example:

Reviewer: ```suggestion
Now in the case of “Enable Logging Only Mode” there will be no clients denied by this feature, meaning the status code will be the expected result (normally 200 but it also depends on your application) and clients will receive the expected normal response, however, the substatus code indicating that one of the two conditions was met will be logged in IIS logs. Below is an example:
```
